custom_components/tuya_custom/tuyaha/devices/climate.py

```diff
@@ -65,21 +65,22 @@ def max_humidity(self):
 
     def set_temperature(self, temperature):
         """Set new target temperature."""
-        self.api.device_control(
-            self.obj_id, "temperatureSet", {"value": float(temperature)}
-        )
+        if self._control_device("temperatureSet", {"value": float(temperature)}):
+            self._update_data("temperature", temperature)
 
     def set_humidity(self, humidity):
         """Set new target humidity."""
         raise NotImplementedError()
 
     def set_fan_mode(self, fan_mode):
         """Set new target fan mode."""
-        self.api.device_control(self.obj_id, "windSpeedSet", {"value": fan_mode})
+        if self._control_device("windSpeedSet", {"value": fan_mode}):
+            self._update_data("windspeed", fan_mode)
 
```

Comment:
I think that here you will receeive "low, medium, high" and should be converted to "1,2,3" (inverted of convertion done in current_fan_mode method)

Reply:
OK, I'm not sure if the parameters given here are "1", "2", "3" or "low", "medium", "high", so I'll change it as follows.

```python
        if self._control_device("windSpeedSet", {"value": fan_mode}):
            fanList = self.fan_list()
            if fan_mode in fanList:
                val = str(fanList.index(fan_mode) + 1)
            else:
                val = fan_mode
            self._update_data("windspeed", val)
```